lokinet/lokinet-base.dockerfile

```diff
@@ -1,24 +1,34 @@
-FROM debian:stable AS lokinet-base
+#use argument instead of lsb-release
+ARG DEBIAN_RELEASE=bullseye 
+
+FROM debian:${DEBIAN_RELEASE}-slim AS lokinet-base
 ENV container docker
+
+ENV RELEASE=${DEBIAN_RELEASE:-bullseye}
+#Add oxen public key 
+ADD --chmod=644 --chown=_apt https://deb.oxen.io/pub.gpg /etc/apt/trusted.gpg.d/lokinet.gpg
+
 # set up packages
-RUN /bin/bash -c 'echo "man-db man-db/auto-update boolean false" | debconf-set-selections'
-RUN /bin/bash -c 'apt-get -o=Dpkg::Use-Pty=0 -q update && apt-get -o=Dpkg::Use-Pty=0 -q dist-upgrade -y && apt-get -o=Dpkg::Use-Pty=0 -q install -y --no-install-recommends ca-certificates curl iptables dnsutils lsb-release systemd systemd-sysv cron conntrack iproute2 python3-pip wget'
-RUN /bin/bash -c 'curl -so /etc/apt/trusted.gpg.d/lokinet.gpg https://deb.oxen.io/pub.gpg'
-RUN /bin/bash -c 'echo "deb https://deb.oxen.io $(lsb_release -sc) main" > /etc/apt/sources.list.d/lokinet.list'
-RUN /bin/bash -c 'apt-get -o=Dpkg::Use-Pty=0 -q update && apt-get -o=Dpkg::Use-Pty=0 -q dist-upgrade -y && apt-get -o=Dpkg::Use-Pty=0 -q install -y --no-install-recommends lokinet'
+# not sure if wget, lsb-release and curl are needed (maybe they can be removed to make a smaller image?)
+RUN DEBIAN_FRONTEND=noninteractive \
+    && echo "deb https://deb.oxen.io ${RELEASE} main" > /etc/apt/sources.list.d/lokinet.list \     
+    && echo "man-db man-db/auto-update boolean false" | debconf-set-selections \
+    && apt update -y \
+    && apt full-upgrade -y \
+    && apt install -y --no-install-recommends ca-certificates iptables dnsutils systemd systemd-sysv cron conntrack iproute2 python3-pip \
+    && apt update -y \    
+    && apt install -y --no-install-recommends lokinet \
+    && apt-get clean \
+    && rm -rf /var/lib/apt/lists/* \
+    && mkdir -p /var/lib/lokinet/conf.d \
+    && mkdir /data && chown _lokinet:_loki /data
 
-# make config dir for lokinet
-RUN /bin/bash -c 'mkdir -p /var/lib/lokinet/conf.d'
-# set up private data dir for lokinet
-RUN /bin/bash -c 'mkdir /data && chown _lokinet:_loki /data'
 
 # print lokinet util
-COPY contrib/print-lokinet-address.sh /usr/local/bin/print-lokinet-address.sh
-RUN /bin/bash -c 'chmod 700 /usr/local/bin/print-lokinet-address.sh'
+COPY --chmod=700 contrib/print-lokinet-address.sh /usr/local/bin/print-lokinet-address.sh
```

Comment:
I realize this is not something changed in this PR, but there is no reason this shouldn't be 755 instead of 700 (it's not doing anything privileged).

Reply:
Why would it need to be 755? If docker's "root" is running it...  why would anyone else require privs?

Reply:
It doesn't need it, but it doesn't *hurt* and it's a more proper permission setting (for instance if someone looks at the recipe for doing a more permanent installation outside a docker container).

Reply:
I was thinking of doing an entrypoint script eventually to allow configuration via environment files/secrets etc. However, limiting privileges and access to secure stuff is pretty important to me. I tend to disagree with 'doesn't hurt' though. IMHO the less stuff is available in those containers, the smaller the attack surface. It would be awesome if we could get rid of python too but I'm still figuring out dependencies.

Reply:
> limiting privileges and access to secure stuff

This script isn't secure in any way: it's just a convenience script to do a DNS query and print the result.  Removing the read/execute permission bit doesn't make anything more secure because anyone could run the same commands already.

Reply:
I don't know tbh. I'll take your word for it.